### PR TITLE
feat(meta): Add catalog in TableMeta

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat(meta): Add catalog in TableMeta

Part of https://github.com/datafuselabs/databend/issues/7816

### CODE REVIEW

**Comment by zhang2014** [comment]
@mergify update

File: src/meta/proto-conv/tests/it/proto_conv.rs [hunk 1]
```diff
@@ -412,57 +413,118 @@ fn test_load_old() -> anyhow::Result<()> {
...
+                202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1,
+                1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99,
+                160, 6, 10, 168, 6, 1,
+            ];
+            let p: pb::TableMeta = common_protos::prost::Message::decode(tbl_meta_v10.as_slice())
+                .map_err(print_err)?;
+
+            let got = mt::TableMeta::from_pb(p).map_err(print_err)?;
+
+            let want = new_table_meta();
```
Comment (drmingdrmer) [inline src/meta/proto-conv/tests/it/proto_conv.rs:525]: There should be a dedicated instance, but not the latest version of table-meta,  for testing this `v10` table-meta.

And using some value other than the default value `"default"` for the compatibility test would be better, to assert that the field `catalog` is loaded from the message but not just filled with a default value.

Comment (Xuanwo) [reply src/meta/proto-conv/tests/it/proto_conv.rs]: Addressed in https://github.com/datafuselabs/databend/pull/7835/commits/c5c12c35e2f47af361a2f6a618b68728fb2ea4b1

**Comment by Xuanwo** [comment]
`tbl_meta_v1` is changed by `rustfmt`, no real content change.

**Review by drmingdrmer — APPROVED** [review]
nice...

**Review by lichuang — APPROVED** [review]

**Pull request merged**